internal/shared/otlp/otlptrace/otlpconfig/options.go.tmpl

```diff
@@ -317,6 +318,13 @@ func WithHeaders(headers map[string]string) GenericOption {
 	})
 }
 
+func WithHostHeader(host string) GenericOption {
```

Comment:
`options_test.go.tmpl` does include tests for other options. Could you add some here too?

Reply:
done